e2etests/tests/node_selector.go

```diff
@@ -0,0 +1,354 @@
+					frr.NeighborConfig.ToAdvertiseV4 = allV4Pfxs
+					frr.NeighborConfig.ToAdvertiseV6 = allV6Pfxs
+				})
+				framework.ExpectNoError(err)
+			}
+
+			err = updater.Update(cfg)
+			framework.ExpectNoError(err)
+
+			for _, c := range frrs {
+				ValidateFRRPeeredWithNodes([]corev1.Node{firstNode}, c, p.ipFamily)
+				ValidateFRRNotPeeredWithNodes(otherNodes, c, p.ipFamily)
+			}
+
+			ginkgo.By("validating")
+			peers := peersV4
+			if p.ipFamily == ipfamily.IPv6 {
+				peers = peersV6
+			}
+
+			pfxsToValidate := p.v4Prefixes
+			globalPfxsToValidate := p.globalv4Prefixes
+			if p.ipFamily == ipfamily.IPv6 {
+				pfxsToValidate = p.v6Prefixes
+				globalPfxsToValidate = p.globalv6Prefixes
+			}
+
+			for _, peer := range peers {
+				ValidateNodesHaveRoutes(pods[:1], peer.FRR, pfxsToValidate...)
+				ValidateNodesDoNotHaveRoutes(pods[1:], peer.FRR, pfxsToValidate...)
+			}
+
+			p.globalModifyPeers(peersV4, peersV6)
+			neighbors = config.NeighborsFromPeers(peersV4, peersV6)
+			globalConfig := frrk8sv1beta1.FRRConfiguration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "testreceive-selector-global",
+					Namespace: "default",
+				},
+				Spec: frrk8sv1beta1.FRRConfigurationSpec{
+					BGP: frrk8sv1beta1.BGPConfig{
+						Routers: []frrk8sv1beta1.Router{
+							{
+								ASN:       p.myAsn,
+								VRF:       p.vrf,
+								Neighbors: neighbors,
+								Prefixes:  globalPfxsToValidate,
+							},
+						},
+					},
+				},
+			}
+			err = updater.Update(globalConfig)
+			framework.ExpectNoError(err)
+
+			for _, c := range frrs {
+				ValidateFRRPeeredWithNodes(allNodes, c, p.ipFamily)
+			}
+
+			for _, peer := range peers {
+				ValidateNodesHaveRoutes(pods[:1], peer.FRR, pfxsToValidate...)
+				ValidateNodesHaveRoutes(pods[:1], peer.FRR, globalPfxsToValidate...)
+				ValidateNodesDoNotHaveRoutes(pods[1:], peer.FRR, pfxsToValidate...)
+				ValidateNodesHaveRoutes(pods[1:], peer.FRR, globalPfxsToValidate...)
+			}
+		},
+			ginkgo.Entry("IPV4", params{
+				ipFamily:         ipfamily.IPv4,
+				vrf:              "",
+				myAsn:            infra.FRRK8sASN,
+				v4Prefixes:       []string{"192.168.2.0/24", "192.169.2.0/24"},
+				globalv4Prefixes: []string{"192.170.2.0/24", "192.171.2.0/24"},
+				modifyPeers: func(ppV4 []config.Peer, _ []config.Peer) {
+					for i := range ppV4 {
+						ppV4[i].Neigh.ToReceive.Allowed.Prefixes = []string{"192.168.2.0/24", "192.169.2.0/24"}
+					}
+				},
+				globalModifyPeers: func(ppV4 []config.Peer, _ []config.Peer) {
+					for i := range ppV4 {
+						ppV4[i].Neigh.ToReceive.Allowed.Prefixes = []string{"192.170.2.0/24", "192.171.2.0/24"}
+					}
+				},
+			}),
+			ginkgo.Entry("IPV6", params{
+				ipFamily:         ipfamily.IPv6,
+				vrf:              "",
+				myAsn:            infra.FRRK8sASN,
+				v6Prefixes:       []string{"fc00:f853:ccd:e799::/64", "fc00:f853:ccd:e800::/64"},
+				globalv4Prefixes: []string{"fc00:f853:ccd:e801::/64", "fc00:f853:ccd:e802::/64"},
```

Comment:
This should have failed the e2etests, I think there's a bug in the `PodHasPrefixFromContainer` function.

Reply:
fixed `v4` -> `v6`, not sure there's a bug since it just didn't validate anything (the `globalPfxsToValidate` became empty since the ipfamily was set to ipv6)